packages/git/src/node/dugite-git-watcher.ts

```diff
@@ -28,19 +29,34 @@ export class DugiteGitWatcherServer implements GitWatcherServer {
             watcher.dispose();
         }
         this.watchers.clear();
+        for (const disposable of this.subscriptions.values()) {
+            disposable.dispose();
```

Comment:
You don't need this after all...

Reply:
True.

Reply:
Wait for a second; who is calling the `dispose` on the `subscriptions`?

Reply:
See this comment: https://github.com/theia-ide/theia/pull/1164#discussion_r164742038

Reply:
`DisposableCollection` gets disposed when all contained disposables are disposed.

Reply:
I would propose documenting this behavior with some spec tests (as a follow-up task), so others who join the development fully understand it. I am not talking about JS-doc, that could get outdated anytime.

Obviously, it was not totally clear to me either. It is very easy to use once you got used to it but clearly, the logic is hidden in the details.

Reply:
One should understand what `DisposableCollection` does and it should be enough. We can add jsdoc for it separately.